src/demo-app/checkbox/checkbox-demo.ts

```diff
@@ -0,0 +1,97 @@
+import {Component} from 'angular2/core';
+import {MdCheckbox} from '../../components/checkbox/checkbox';
+
+interface Task {
+  name: string,
+  completed: boolean,
+  subtasks?: Task[]
+}
+
+@Component({
+  selector: 'md-checkbox-demo-nested-checklist',
+  styles: [`
+    li {
+      margin-bottom: 4px;
+    }
+  `],
+  template: `
+    <h2>Tasks</h2>
```

Comment:
I'm much more on the side of separate files for everything; I generally find it easier to get overwhelmed when everything is all in one. 

I would say to go for separate files here, though, mainly because it's more toolable (i.e., able to glob all html files).